pkg/datapath/linux/routing/routing.go

```diff
@@ -260,3 +260,24 @@ func retrieveIfIndexFromMAC(mac mac.MAC, mtu int) (index int, err error) {
 func ComputeTableIDFromIfaceNumber(num int) int {
 	return linux_defaults.RouteTableInterfacesOffset + num
 }
+
+// RetrieveIfaceNameFromMAC finds the corresponding device name for a
+// given MAC address.
+func RetrieveIfaceNameFromMAC(mac string) (devName string, err error) {
+	var links []netlink.Link
+
+	links, err = netlink.LinkList()
+	if err != nil {
+		err = fmt.Errorf("unable to list interfaces: %s", err)
+		return
+	}
+	for _, l := range links {
+		if l.Attrs().HardwareAddr.String() == mac {
+			devName = l.Attrs().Name
+			return
+		}
+	}
+
+	err = fmt.Errorf("interface with MAC %s not found", mac)
```

Comment:
ideally, errors should not include dynamic details like endpoint-specific state in them. I think we can move the `mac` piece into the `log.WithFields(...)` in the caller. This structured logging allows log handling software to better track & collapse log messages because if this error occurs for multiple MACs, then the error message will be consistent across all of them.

Reply:
The naunce in this code though is that it'll be in the CNI code path, so the logs won't be in the agent, but rather in kubelet. Given that, I'd be inclined to staying with dynamic formatting. WDYT?

Reply:
In this particular case, I think this is new functionality only called by the Cilium-agent code right?

I agree though if the error is going through cilium-cni out to kubelet then that's a reasonable argument to include dynamic details directly in the error messages.

Reply:
Ha, you're totally right :sweat_smile:. I've just had `pkg/datapath/linux/routing/routing.go` hardwired in my head to be called from CNI code